### DISCUSSION TITLE
Keep Container Alive for Kafka Trigger

### DISCUSSION THREAD

**Discussion by jerdmann3** [kedacore/keda#4210] [General]
When using the Kafka trigger with minReplicaCount = 0, is there a way to keep a pod alive during constant activity on the consuming topic? Currently it seems the behavior is to activate a pod when lag is detected on the broker, spin up a pod and consume the lag, then shut down that pod after the cooldownPeriod. This seems to create a constant cycle of deploying and terminating a pod during constant topic activity but it would be nice to keep that container alive to avoid this cycle. Is there a way to do that currently?

**Comment by zroubalik** [comment]
try setting `minReplicaCount = 1`

**Reply by jerdmann3** [reply]
Sure, but I want to take advantage of the scale to 0 capability during times of 0 activity. But it seems like in order to do this, you need to accept the stop/start cycle I mentioned above during times of constant activity. It would be nice if there was a way to reset the cooldown timer when a new Kafka messages arrives to keep the container alive and avoid instantly terminating when cooldownPeriod has been reached.

**Reply by zroubalik** [reply]
I am sorry, but I am not sure if I fully understand the problem. If there's any activity happening during cooldown period then is this period canceled. So maybe extending this period might help you?
